addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -9155,39 +9161,38 @@ QUnit.module("Views", (hooks) => {
         // In this test, we try to reproduce the deadlock situation by forcing the field widget to
         // commit changes before the save. We thus manually call 'saveRecord', instead of clicking
         // on 'Save'.
-        assert.expect(6);
+        assert.expect(7);
 
-        await makeView({
+        const form = await makeView({
             type: "form",
             resModel: "partner",
             serverData,
             arch: '<form><field name="foo"/></form>',
             mockRPC(route, args) {
                 assert.step(args.method);
-                return this._super.apply(this, arguments);
             },
             // we set a fieldDebounce to precisely mock the behavior of the webclient: changes are
             // not sent to the model at keystrokes, but when the input is left
-            fieldDebounce: 5000,
+            // fieldDebounce: 5000,
         });
 
-        await editInput(target.querySelector(".o_field_widget[name=foo] input"), "some foo value");
+        await editInput(target, ".o_field_widget[name=foo] input", "some foo value");
```

Comment:
as there was a field debounce, I think you can't use editInput here (which will trigger a change event). Instead, you'll have to manually set the input value and trigger an "input" event. That way, the field hasn't notify the change to the model yet, when you save the record